packages/formstr-app/src/formLifecycle.test.ts

```diff
@@ -0,0 +1,189 @@
+import { Event, SimplePool, generateSecretKey, getPublicKey } from 'nostr-tools';
+import { bytesToHex } from '@noble/hashes/utils';
+import { createForm } from './nostr/createForm';
+import { sendResponses } from './nostr/common';
+import { fetchFormResponses } from './nostr/responses';
+import { Response, Tag } from './nostr/types';
+import { execSync } from 'child_process';
+
+const TEST_RELAYS = [`http://localhost:3000/`];
+const DOCKER_IMAGE_NAME = 'billa05/test-relay:latest';
+
+jest.setTimeout(200000);
+
+describe('Form Lifecycle Behavioral Test', () => {
+  let pool: SimplePool;
+  let formSecretKey: string;
+  let formPublicKey: string;
+  let formId: string;
+  let containerId: string;
+
+  beforeAll(async () => {
+    console.log('Starting relay server with Docker...');
+
+    try {
+      console.log(`Pulling Docker image ${DOCKER_IMAGE_NAME}...`);
+      execSync(`docker pull ${DOCKER_IMAGE_NAME}`, { stdio: 'inherit' });
+      console.log(`Successfully pulled image: ${DOCKER_IMAGE_NAME}`);
+
+      const containerName = `nostr-relay-test-${Date.now()}`;
+      const startCmd = `docker run -d --name ${containerName} -p 3000:3000 ${DOCKER_IMAGE_NAME}`;
```

Comment:
should we use a different port, since this is also the port for the dev server, it shouldn't conflict on the test env, but let's not leave that possibility open